File: src/FinAnalyzer.Domain/Models/RawTransaction.cs
namespace FinAnalyzer.Domain.Models;

public sealed class RawTransaction
{
    public required Guid Id { get; init; }
    public required DateOnly Booking { get; init; }
    public required DateOnly ValueDate { get; init; }
    public required string PayerOrPayee { get; init; }
    public required string BookingText { get; init; }
    public required string Purpose { get; init; }
    public Currency? Balance { get; init; }
    public required Currency Amount { get; init; }
}

File: frontend/src/models/dtos/raw-transaction.ts
import { Currency } from "./currency";


export interface RawTransaction {
    id: string;
    booking: string; // DateOnly is typically represented as a string (ISO format)
    valueDate: string;
    payerOrPayee: string;
    bookingText: string;
    purpose: string;
    balance?: Currency;
    amount: Currency;
}
